var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -176,6 +178,29 @@ def install(self, spec, prefix):
                         '-rhsone')
             make("install")
 
+    @run_after('install')
+    def cache_test_sources(self):
+        srcs = ['src/examples']
+        self.cache_extra_test_sources(srcs)
+
+    def test(self):
+        """Perform smoke test on installed HYPRE package."""
+
+        if '+mpi' in self.spec:
+            examples_dir = join_path(self.install_test_root, 'src/examples')
+            with working_dir(examples_dir, create=False):
+                make("HYPRE_DIR=" + self.prefix, "bigint")
+
+                reason = "test: ensuring HYPRE examples run"
+                self.run_test('./ex5big', [], [], installed=True,
+                              purpose=reason, skip_missing=True, work_dir='.')
+                self.run_test('./ex15big', [], [], installed=True,
+                              purpose=reason, skip_missing=True, work_dir='.')
+
+                make("distclean")
+        else:
+            raise RuntimeError("MPI is required to run HYPRE smoke test")
```

Comment:
I think it's better to remove this. We don't want the tests to fail for `hypre~mpi`.

Reply:
I agree.  It would be better to "skip" the test if it does not apply.